towhee/runtime/nodes/_filter.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .node import Node
+from towhee.runtime.data_queue import ColumnType
+
+class Filter(Node):
+    """
+    Filter Operator.
+
+    Filter the input columns based on the selected filter_columns and filter.
+    """
+    def process_step(self) -> bool:
+        assert len(self._node_repr.outputs) == len(self._node_repr.inputs)
+
+        data = self._in_ques[0].get_dict()
+        if data is None:
+            self._set_finished()
+            return True
+
+        process_data = [data.get(key) for key in self._node_repr.iter_info.param['filter_by']]
+        succ, outputs, msg = self._call(process_data)
+        if not succ:
+            self._set_failed(msg)
+            return True
+
+        if outputs:
+            output_map = {new_key: data[old_key] for new_key, old_key in zip(self._node_repr.outputs, self._node_repr.inputs)}
+            data.update(output_map)
+
+        scalar = [self._in_ques[0].schema[i] for i in range(len(self._in_ques[0].schema)) if self._in_ques[0].type_schema[i] == ColumnType.SCALAR]
+
+        for out_que in self._output_ques:
+            inter = set(out_que.schema) - set(scalar) & set(list(data.keys()))
```

Comment:
`in_queue` : DataQueue([('url', SCALAR)])
`out_queue`: DataQueue([('url', SCALAR)]) 
it seems nothing will be put into `out_queue`